### PR TITLE
Fix student run of copied AP activity opens default AP activity

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/179378800

[#179378800]

The `activity` param was being inadvertently stripped out of at least some Activity Player activity URLs when student users tried to run the activities. This change ensures that the `activity` param is kept in the URL.

### CODE REVIEW

**Comment by emcelroy** [comment]
@scytacki I went ahead with option two and removed `launch_url`. I struggled a bit with the automated tests, though. I think they're OK now, but I may have altered the test URLs more than necessary. I'm not sure.

Also, there's a reference to `launch_url` in rails/lib/tasks/update_external_activity.rake. I wasn't sure if I should update that. Seems like it's sort of an outdated script that won't need to be run going forward. Would it be best to update it anyway?

**Comment by scytacki** [comment]
@emcelroy you might as well delete the rake task. I agree it won't be used again and if it is, it isn't hard to recreate it.

File: rails/app/controllers/portal/offerings_controller.rb [hunk 1]
```diff
@@ -58,8 +58,9 @@ def show
            # session[:put_path] = nil
          end
          external_activity = @offering.runnable
-         if external_activity.launch_url.present?
-           uri = URI.parse(external_activity.launch_url)
+         tool = Tool.where(id: external_activity.tool_id).first
```
Comment (scytacki) [inline rails/app/controllers/portal/offerings_controller.rb:61]:
```suggestion
         tool = external_activity.tool
```

File: rails/app/models/external_activity.rb [hunk 2]
```diff
@@ -299,20 +295,13 @@ def run_format
...
-  end
-
-  def lara_sequence?
-    material_type == 'Investigation' && has_launch_url?
-  end
-
   def lara_activity_or_sequence?
-    lara_activity? || lara_sequence?
+    tool = Tool.where(id: self.tool_id).first
+    if tool && tool.source_type == 'LARA'
```
Comment (scytacki) [inline rails/app/models/external_activity.rb:300]:
```suggestion
    if self.tool && self.tool.source_type == 'LARA'
```

The external activity models defines a `belongs_to :tool` association. So you can just call self.tool (or if you want to be more concise you can drop the `self.`)

**Comment by emcelroy** [comment]
@scytacki OK, I've deleted update_external_activity.rake.

File: rails/spec/models/external_activity_spec.rb [hunk 3]
```diff
@@ -258,25 +256,48 @@
...
+        :url => "http://www.concord.org/ap",
+        :template_type => "Activity",
+        :tool_id => ap_tool.id
+    } }
+    let(:lara_activity) { ExternalActivity.create!(lara_attributes)}
+    let(:ap_activity) { ExternalActivity.create!(ap_attributes)}
 
-    it "activities with launch urls should return true for lara_activity?" do
-      expect(activity.lara_activity?).to be true
+    it "activities with LARA as its tool should return true for lara_activity?" do
```
Comment (scytacki) [inline rails/spec/models/external_activity_spec.rb:296]:
```suggestion
    it "activities with LARA as its tool should return true for lara_activity_or_sequence?" do
```

File: rails/spec/models/external_activity_spec.rb [hunk 4]
```diff
@@ -258,25 +256,48 @@
...
+      expect(lara_activity.lara_activity_or_sequence?).to be true
     end
-    it "activities wihtout launch urls should return false for lara_activity?" do
-      activity.launch_url = nil
-      expect(activity.lara_activity?).to be false
+    it "activities with something other than LARA as a tool should return false for lara_activity?" do
```
Comment (scytacki) [inline rails/spec/models/external_activity_spec.rb:299]:
```suggestion
    it "activities with something other than LARA as a tool should return false for lara_activity_or_sequence?" do
```

**Review by scytacki — CHANGES_REQUESTED** [review]
Mostly looks good to me. I just found a few minor changes to use the existing tool relationship on the ExternalActivity model.  And some test descriptions not updated.

File: rails/app/controllers/portal/offerings_controller.rb (hunk 1 above)
Comment (emcelroy) [reply rails/app/controllers/portal/offerings_controller.rb]: Thanks, @scytacki. I did this change a little differently. Does what I did make sense? It seems to work fine.

**Review by scytacki — APPROVED** [review]
looks good to me

**Pull request merged**